src/header.rs

```diff
@@ -242,11 +243,13 @@ impl CocoonHeader {
         salt.copy_from_slice(&buf[8..24]);
         let mut nonce = [0u8; 12];
         nonce.copy_from_slice(&buf[24..36]);
-        let length = usize::from_be_bytes(
-            buf[36..Self::SIZE]
-                .try_into()
-                .map_err(|_| Error::TooLarge)?,
-        );
+
+        let length = {
+            let bytes = &buf[36..Self::SIZE];
+            let bytes = <[u8; 8]>::try_from(bytes).map_err(|_| Error::TooLarge)?;
+            let length = u64::from_be_bytes(bytes);
+            usize::try_from(length).map_err(|_| Error::TooLarge)?
+        };
```

Comment:
Yeah, looks sad. Maybe we can do it a bit cleaner at least:
```rust
let mut bytes = [0u8; 8];
bytes.copy_from_slice(&buf[36..Self::SIZE]);
let length = u64::from_be_bytes(bytes);
usize::try_from(length).map_err(|_| Error::TooLarge)?
```
Agree?

Reply:
And I think we are almost done overall 👍

Reply:
I've rewritten the old `try_into` to `<[u8; 8]>::try_from` to handle the case when `buf[36..Self::SIZE]` would be longer than 8 bytes. If it always be 8 bytes, then your new code is better.

But then, there is a question - why we need the `Self::SIZE` at all? We would write just `buf[36..44]` and done, if I don't overlook something?

Reply:
@MartinKavik Yeah, basically, as I remember, I need SIZE in a few places I wanted to easily change the SIZE in a case of refactoring getting the compilation error if I'm messed up with sizes at some point. So, I understand that in this concrete case it looks a bit unclear, but let's not remove the SIZE for now. :) (Honestly, I played around with that SIZE for a couple of evenings, because I stumbled upon different non-obvious things during coding and refactoring, and finally decided to stick with it, so just no need to touch it 😅).

Reply:
Thanks for the explanation.
Refactored & force-pushed. I think I see the finish line.. :)